lib/rom/repository/command_proxy.rb

```diff
@@ -11,11 +11,10 @@ class CommandProxy
 
       def initialize(command)
         @command = command
-        @root = Dry::Core::Inflector.singularize(command.name.relation).to_sym
       end
 
       def call(input)
-        command.call(root => input)
+        command.call(command.name.relation => input)
       end
```

Comment:
@jasper-lyons yeah it crossed my mind to just allow `nil` but it's a bit nasty, because we'd be adding special meaning to `nil` and IF it's `nil` when it shouldn't be we'd be wondering why and debugging would be harder. So I'd say it's better to use input evaluator when there's actually something to evaluate, otherwise just pass the input as it is.

> I don't understand how the path is built up though.

It's built by traversing through the command graph ast and building it up as we dive deeper, see [the spec](https://github.com/rom-rb/rom/blob/master/spec/integration/commands/graph_spec.rb#L94-L119) where `option` is a sample ast, we have `:users` then `:tasks` so we end up with `:'users.tasks'` tuple path while traversing through these options.